azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -69,41 +82,65 @@ func resourceArmEventGridEventSubscription() *schema.Resource {
 				}, false),
 			},
 
+			"expiration_time_utc": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
 			"topic_name": {
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
 			},
 
-			"storage_queue_endpoint": {
+			AzureFunctionEndpointID: {
```

Comment:
Could we leave these as
```suggestion
			"azure_function_endpoint_id: {
```

it becomes hard to read the schema when we abstract this out

Reply:
After completing the refactoring, thoughts came in the same direction. Indeed, (schema) key abstraction is not a good coding practice. I'll change this immediately.